pkg/application-grid-wrapper/storage/cache.go

```diff
@@ -135,11 +141,38 @@ func (sc *storageCache) GetNode(hostName string) *v1.Node {
 	return nil
 }
 
+func (sc *storageCache) GetLocalNodeInfo() map[string]data.ResultDetail {
+	sc.mu.RLock()
+	defer sc.mu.RUnlock()
+	return sc.localNodeInfo
+}
+
+func (sc *storageCache) ClearLocalNodeInfo() {
+	sc.mu.Lock()
+	defer sc.mu.Unlock()
+	sc.localNodeInfo = make(map[string]data.ResultDetail)
+}
+
+func (sc *storageCache) SetLocalNodeInfo(info map[string]data.ResultDetail) {
+	klog.V(4).Infof("Set local node info %#v", info)
+	sc.mu.Lock()
+	sc.localNodeInfo = info
+
+	// update endpoints
+	changedEps := sc.rebuildEndpointsMap()
+
+	sc.mu.Unlock()
+
+	for _, eps := range changedEps {
+		sc.endpointsChan <- eps
+	}
+}
+
 // rebuildEndpointsMap updates all endpoints stored in storageCache.endpointsMap dynamically and constructs relevant modified events
 func (sc *storageCache) rebuildEndpointsMap() []watch.Event {
 	evts := make([]watch.Event, 0)
 	for name, endpointsContainer := range sc.endpointsMap {
-		newEps := pruneEndpoints(sc.hostName, sc.nodesMap, sc.servicesMap, endpointsContainer.endpoints, sc.wrapperInCluster)
+		newEps := pruneEndpoints(sc.hostName, sc.nodesMap, sc.servicesMap, endpointsContainer.endpoints, sc.localNodeInfo, sc.wrapperInCluster, sc.serviceAutonomyEnhancementSwitch)
 		if apiequality.Semantic.DeepEqual(newEps, endpointsContainer.modified) {
 			continue
 		}
```

Comment:
please add lock for operation ”sc.endpointsMap[name].modified = newEps“

Reply:
Ok, i will create another PR to fix this.